### PR TITLE
Revert "rgw: register daemon in service map with more details"

### PR DESCRIPTION
**Pull request by liewegas**
This reverts commit 46ec2f0ddd81b4b5b4a61633eada998dcd606c9b.

This break cephadm (by triggering CEPHADM_STRAY_DAEMON) because it
assumes that a daemon named rgw.r.z.foo will register as rgw.r.z.foo.
It is not clear to me that there is a way to work around this disparity
that makes much sense.  I think it makes more sense to focus on the
use-case that needs daemons to register under unique names and perhaps
control that behavior via an option or invest in providing daemons with
unique ids.

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Review by sebastian-philipp — APPROVED** [review]
Fixes https://pulpito.ceph.com/swagner-2021-03-05_12:08:19-rados:cephadm-wip-swagner3-testing-2021-03-05-1110-distro-basic-smithi/ 

See #39380

**Comment by mattbenjamin** [comment]
@liewegas cephadm should not be preventing us from registering daemons with unique, and intelligible names.

**Comment by liewegas** [comment]
> @liewegas cephadm should not be preventing us from registering daemons with unique, and intelligible names.

Right--the problem with the change is that radosgw is preventing cephadm from using unique, intelligible names, by appending a random value to what we provide.  cephadm says "please use name X" and radosgw starts up as X.$something. 

> If the name cephadm things the instance should have is already satisfactory, then we should be able to use that. If the name cephadm things the instance should have is not unique, then I think we need a systemic improvement. The high level objectives I think we're trying to serve are both a) allow a very simple, clean implementation of instance scaling by k8s hpa (don't need to generate and provision cephx keys for every transient pod created to scale-up the object storage service); and b) provide for intelligible counters and counter rollup in mgr/dashboard, including for multisite sync clusters

Agree on objectives.  I think we should focus on the cases where radosgw gw is starting up with a non-unique name (and why) and ensure that case can achieve those goals.  E.g., making the random suffix option and opt-in... or avoid registering in the service map at all (if it's ephemeral)...

File: qa/suites/rgw/singleton/all/rgw_cephadm.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+roles: [mon.a, osd.0, osd.1, osd.2, rgw.z.r.a, rgw.z.r.b]
+tasks:
+- cephadm:
+- exec:
+    mon.a:
+    - sleep 30
```
Comment (cbodley) [inline qa/suites/rgw/singleton/all/rgw_cephadm.yaml:6]: it doesn't look like anybody tested this before merging. it's been failing consistently in teuthology/task/kernel.py, but i don't understand the connection to kernel.py
```
2021-04-12T20:22:31.262 DEBUG:teuthology.orchestra.run.smithi033:> rpm -q kernel --last | head -n 1
2021-04-12T20:22:31.394 INFO:teuthology.orchestra.run.smithi033.stdout:kernel-4.18.0-240.22.1.el8_3.x86_64           Mon 12 Apr 2021 08:22:08 PM UTC
2021-04-12T20:22:31.396 DEBUG:teuthology.task.kernel:get_latest_image_version_rpm: 4.18.0-240.22.1.el8_3.x86_64
2021-04-12T20:22:31.396 INFO:teuthology.task.kernel:Not newest distro kernel. Current: 4.18.0-240.10.1.el8_3.x86_64 Expected: 4.18.0-240.22.1.el8_3.x86_64
2021-04-12T20:22:31.397 ERROR:teuthology.run_tasks:Saw exception from tasks.
Traceback (most recent call last):
  File "/home/teuthworker/src/git.ceph.com_git_teuthology_2713a3cd31b17738a50039eaa9d859b5dc39fb8a/teuthology/run_tasks.py", line 91, in run_tasks
    manager = run_one_task(taskname, ctx=ctx, config=config)
  File "/home/teuthworker/src/git.ceph.com_git_teuthology_2713a3cd31b17738a50039eaa9d859b5dc39fb8a/teuthology/run_tasks.py", line 70, in run_one_task
    return task(**kwargs)
  File "/home/teuthworker/src/git.ceph.com_git_teuthology_2713a3cd31b17738a50039eaa9d859b5dc39fb8a/teuthology/task/kernel.py", line 1247, in task
    (role_remote,) = ctx.cluster.only(role).remotes.keys()
ValueError: too many values to unpack (expected 1)
2021-04-12T20:22:31.658 ERROR:teuthology.run_tasks: Sentry event: https://sentry.ceph.com/organizations/ceph/?query=a96ff9124af145a5a334f39083445e0c
Traceback (most recent call last):
  File "/home/teuthworker/src/git.ceph.com_git_teuthology_2713a3cd31b17738a50039eaa9d859b5dc39fb8a/teuthology/run_tasks.py", line 91, in run_tasks
    manager = run_one_task(taskname, ctx=ctx, config=config)
  File "/home/teuthworker/src/git.ceph.com_git_teuthology_2713a3cd31b17738a50039eaa9d859b5dc39fb8a/teuthology/run_tasks.py", line 70, in run_one_task
    return task(**kwargs)
  File "/home/teuthworker/src/git.ceph.com_git_teuthology_2713a3cd31b17738a50039eaa9d859b5dc39fb8a/teuthology/task/kernel.py", line 1247, in task
    (role_remote,) = ctx.cluster.only(role).remotes.keys()
ValueError: too many values to unpack (expected 1)
```

i raised https://tracker.ceph.com/issues/50338 against the teuthology project

**Pull request merged**